phobos_umss.sh

```diff
@@ -0,0 +1,490 @@
+    error_5=$?
+    save_log rm "$ph_del_script return with status=$ph_del($error_5)"
+    
+    if [ "$error_5" != 0 ]; then
+        STATUS="FAILURE"
+        save_log rm "STATUS=$STATUS($error_5)"
+    fi
+    save_log rm "END rm($*)"
+    return $error_5
+}
+
+# function to rename a file $1 into $2 in the MSS
+# phobos doesn't have the ability to rename a file, or to change its location in its database.
+# The only way currently is to copy $1 into a new object $2, which can be done in two steps :
+# - GET the object $1 into a /tmp/phobos_mv file ;
+# - PUT the file /tmp/phobos_mv into an object $2 ;
+# - DEL the object $1 from phobos.
+mv () {
+    # <your command to rename a file in the MSS> $1 $2
+    # e.g: /usr/local/bin/rfrename rfioServerFoo:$1 rfioServerFoo:$2
+    
+    save_log mv "BEGIN mv($*)"
+    op_ph=$(type -P phobos)
+    op_sys=$(type -P rm)
+    temp="/tmp/phobos_mv"
+
+    # GET the object $1 into a temporary file.
+    ph_get_script="sudo $op_ph get ${1} ${temp}"
+    ph_get=$($ph_get_script >> $LOGFILE_FANCY 2>&1)
+    error_6=$?
+    save_log mv "$ph_get_script return with status=$ph_get($error_6)"
+    if [ $error_6 == 0 ]; then
+        # PUT the file /tmp/phobos_mv into an object $2.
+        ph_put_script="sudo $op_ph put -f dir ${temp} ${2}"
+        ph_put=$(ph_put_script >> $LOGFILE_FANCY 2>&1)
+        error_6=$?
+        save_log mv "$ph_put_script return with status=$ph_put($error_6)"
+        if [ $error_6 == 0 ]; then
+            # DEL the object $1 from phobos.
+            ph_del_script="sudo $op_ph del ${1}"
+            ph_del=$($ph_del_script >> $LOGFILE_FANCY 2>&1)
+            error_6=$?
+            save_log mv "$ph_del_script return with status=$ph_del($error_6)"
+            if [ $error_6 == 0 ]; then
+                rm_temp_script="sudo $op_sys -f ${temp}"
+                rm_temp=$($rm_temp_script) # TODO Trouver un moyen plus sécuritaire serait utile.
+                error_6=$?
+                save_log mv "$rm_temp return with status=$rm_temp($error_6)"
+            fi
+        else # In case of failure, we still need to remove the temporary copy.
+            rm_temp_script="sudo $op_sys -f ${temp}"
+            rm_temp=$($rm_temp_script) # TODO Trouver un moyen plus sécuritaire serait utile.
+            error_6=$?
+            save_log mv "$rm_temp return with status=$rm_temp($error_6)"
+        fi
+    else # In case of failure, we still need to remove the temporary copy, (if it happened to be created).
+        if [ -f "$2" ]; then
+            rm_temp_script="sudo $op_sys -f ${temp}"
+            rm_temp=$($rm_temp_script) # TODO TYrouver un moyen plus sécuritaire serait utile.
+            error_6=$?
+            save_log mv "$rm_temp return with status=$rm_temp($error_6)"
+        fi
+    fi
+
+    if [ $error_6 != 0 ]; then
+        STATUS="FAILURE"
+        save_log mv "STATUS=$STATUS($error_6)"
+    fi
+    save_log mv "END mv($*)"
+    return $error_6
+}
```

Comment:
major: need to be rewritten with a true phobos mv function

Reply:
In the next commit, the code is refactored, and the future code for Phobos rename is in place (but commented, waiting for phobos_rename to be operational).